### PR TITLE
Add tests to diesel example

### PR DESCRIPTION
**Pull request by danitrod**
To facilitate the life of new diesel users who use TDD

_Closes #409_

### CODE REVIEW

**Comment by danitrod** [comment]
hmm CI test fails cause there is no sqlite.. I'll try to work on something

**Comment by danitrod** [comment]
ok, managed to do it with a custom github action to run diesel migrations, hope its ok to add another agent to the workflow

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        version: ["stable", "nightly"]
+        version: ['stable', 'nightly']
```
Comment (JohnTitor) [inline .github/workflows/linux.yml:18]: Why did you change this?

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -49,6 +49,13 @@ jobs:
         with:
           redis-version: 6
 
+      - name: run diesel migrations
+        uses: danitrod/diesel-cli-action@v0.3.4
+        with:
+          migration_directories: database_interactions/diesel
+          database_url: test.db
+          sqlite: true
```
Comment (JohnTitor) [inline .github/workflows/linux.yml:57]: Looking at https://github.com/ducharmemp/diesel-cli-action, I don't think it's complicated enough to use an external action. I'd like to hear some thoughts from @robjtede.

Comment (robjtede) [reply .github/workflows/linux.yml:57]: yes, please inline this action

File: .github/workflows/linux.yml (hunk 1 above)
Comment (danitrod) [reply .github/workflows/linux.yml:18]: Sorry, I think my prettier changed it automatically, will revert

File: .github/workflows/linux.yml (hunk 2 above)
Comment (danitrod) [reply .github/workflows/linux.yml:57]: Alright, will try to inline it later today

**Review by robjtede — CHANGES_REQUESTED** [review]

File: database_interactions/diesel/README.md [hunk 3]
```diff
@@ -26,11 +26,18 @@ brew install sqlite3
 cd examples/diesel
 cargo install diesel_cli --no-default-features --features sqlite
 
-echo "DATABASE_URL=test.db" > .env
+echo "DATABASE_URL=prod.db" > .env
```
[the remark this answers is not in the document]
Comment (danitrod) [reply database_interactions/diesel/README.md:29]: Though of using a separate db for tests and prod, then if doing that the test db should be the one called `test.db` - does that make sense?

[the remark this answers is not in the document]
Comment (robjtede) [reply database_interactions/diesel/README.md:29]: I think given that the db files don't exist there is no difference, the test database in CI is ephemeral. Also don't want to give the impression to readers that this example is suitable for production use, as sqlite often isn't.

[the remark this answers is not in the document]
Comment (danitrod) [reply database_interactions/diesel/README.md:29]: true, I was thinking more as kind of a good practice, but you have a point, this shouldn't be seen as production use. i'll use the same db for tests then

**Review by robjtede — APPROVED** [review]

**Review by JohnTitor — APPROVED** [review]
LGTM, thanks!
